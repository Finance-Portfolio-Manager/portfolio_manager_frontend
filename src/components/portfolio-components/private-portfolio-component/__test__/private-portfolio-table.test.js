import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import PrivatePortfolioTable from './../private-portfolio-table';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { Switch, Route, BrowserRouter as Router } from 'react-router-dom';

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

const tableRows = [{symbol: "TSLA", currentPrice: 5}, {symbol: "GME", currentPrice: 100}]

const toggleFn = jest.fn();

configure({ adapter: new Adapter() });
const wrapper = shallow(<PrivatePortfolioTable tableRows={tableRows} toggleShowDetails={toggleFn}></PrivatePortfolioTable>);

test("matches snapshot", ()=>{
    const tree = renderer.create(<PrivatePortfolioTable tableRows={tableRows} toggleShowDetails={toggleFn}></PrivatePortfolioTable>).toJSON();
    expect(tree).toMatchSnapshot();
})