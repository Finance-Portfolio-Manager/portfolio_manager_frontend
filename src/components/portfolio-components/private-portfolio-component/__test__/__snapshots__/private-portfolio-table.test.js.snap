// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="wrapper fadeInDown folioContent mx-3"
  id="portfolio"
>
  <div
    id="portfolio-values"
  >
    <h4
      id="portfolio-total"
    />
    <div
      id="portfolio-change"
    />
    <div
      id="dollar-change"
    />
  </div>
  <div
    className="table-responsive secondary-text"
  >
    <div
      className="table-responsive"
    >
      <table
        className="table borderless table-hover secondary-text table"
        id="stocks-view"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              Stock
            </th>
            <th
              scope="col"
            >
              Quantity
            </th>
            <th
              scope="col"
            >
              Average Buy Price
            </th>
            <th
              scope="col"
            >
              Current Price
            </th>
            <th
              scope="col"
            >
              P/L
            </th>
          </tr>
        </thead>
        <tbody
          id="stocks-body"
        >
          <tr>
            <td>
              TSLA
            </td>
            <td />
            <td />
            <td>
              5
            </td>
            <td />
          </tr>
          <tr>
            <td>
              GME
            </td>
            <td />
            <td />
            <td>
              100
            </td>
            <td />
          </tr>
        </tbody>
      </table>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="button"
    >
      Hide Details
    </button>
  </div>
</div>
`;
